### PR TITLE
Initial take on official new module template.

### PR DESCRIPTION
**Pull request by rkeithhill**
OK I don't like that we have three copies of build.psake.ps1 but will address later.

### CODE REVIEW

**Comment by rkeithhill** [comment]
This is a start.  Definitely not done yet.

File: src/Templates/NewModule/Tests/Module.T.ps1 [hunk 1]
```diff
@@ -0,0 +1,10 @@
+$ModuleManifestName = '<%=$PLASTER_PARAM_ModuleName%>.psd1'
+# <%=${PLASTER_GUID1}%> - testing use of PLASTER predefined variables.
```
Comment (daviwil) [inline src/Templates/NewModule/Tests/Module.T.ps1]: Probably should get rid of this line?

File: src/Templates/NewModule/build.psake.ps1 [hunk 2]
```diff
@@ -0,0 +1,565 @@
+##############################################################################
+# PREVIEW VERSION OF PSAKE SCRIPT FOR MODULE BUILD & PUBLISH TO THE PSGALLERY
+##############################################################################
+#
+# We are hoping to add support for publishing modules to the PowerShell gallery
```
Comment (daviwil) [inline src/Templates/NewModule/build.psake.ps1]: Probably need to update the comments here to remove the language about our plans

File: src/Templates/NewModule/Tests/Module.T.ps1 (hunk 1 above)
Comment (rkeithhill) [inline src/Templates/NewModule/Tests/Module.T.ps1]: Yup. Thanks for finding that.

File: src/Templates/NewModule/build.psake.ps1 (hunk 2 above)
Comment (rkeithhill) [inline src/Templates/NewModule/build.psake.ps1]: There's a bit of cleanup to do on the build script.

**Comment by rkeithhill** [comment]
> duplicate build script issue
I'm going to get rid of one `examples\NewModuleTemplate` since `src\Templates\NewModule` effectively replaces it.  I think in `examples` we want a simple template that shows various features, some of which are not used in the New Module template.  That template should not be related to creating a new module in order to avoid confusion.  

WRT build.psake.ps1 I was hoping we could achieve something like MSBuild's Microsoft.CSharp.targets file where we would provide most of the standard "build" pipeline (copy appropriate files, code-sign, test, docgen, publish) where the psake script would gracefully handle scenarios where you don't have a code signing cert or don't want to publish.  Additionally, the script needs to have to appropriate, simple extension points like the PrePublish/PostPublish tasks.  We build all sorts of C# projects but we've never had to modify Microsoft.CSharp.targets.  Not quite there with the psake script but the bulk of what needs to be tweaked is at the top of the file in the form of various variables.

**Comment by daviwil** [comment]
Yep, definitely, having a standard build script that doesn't need to be touched by anyone would be nice.  If we can find a way to have an optional properties/tasks file which can be loaded up by convention in build.psake.ps1 that might work pretty well for build extensibility.

**Pull request merged**